SGA-128-Friend-invites

GillWell opened the pull request:
<!-- Enter a brief description/summary of your PR here. What does it fix/what does it change/how was it tested (even manually, if necessary)? -->
## Summary of the Pull Request

SGA 128
Create a friend invite screen to show received friend requests, functions for accepting, declining, sending friend requests.

SGA 129
Create a screen to show current friends, functions to get the friends data and remove them

<!-- Please review the items on the PR checklist before submitting-->
## PR Checklist

SGA 128
- [x] “Friends” screen is accessed from avatar click dropdown
- [x] Friend invite functions should be displayed inside a tab “Friend invites” within “Friends” screen
- [x] Friend invites tab should also display received invites list
- [x] Received invites can be accepted or denied
- [ ] (optional) User should be notified when he receives a friend invite (either through a system notification or a bubble near the icon where friends screen is located (see example below))

SGA 129
- [x] Friends list should be displayed inside a tab “Friends list” within “Friends” screen

<!-- Provide a more detailed description of the PR, other things fixed or any additional comments/features here -->
## Detailed Description of the Pull Request / Additional comments
To send a friend request the user needs to use the search bar to first find the user then send a friend request. User who have sent a friend request will be showed below, the user can accept or decline. Can't send friend request to yourself, friends, users that have sent a friend request already. Friend list shows users that you are friends with. Able to remove friends

GillWell committed:
Friend invites

GillWell committed:
🤖 Apply Spotless

mantasjasikenas commented on app/src/main/res/values/strings.xml:
```diff
@@ -179,4 +179,21 @@
+    <string name="received_friend_requests">Received friend requests</string>
+    <string name="other_users_who_have_sent_you_friend_requests">Other users who have sent you friend requests</string>
+    <string name="search_for_a_user_to_display_them_here">Search for a user to display them here.</string>
+    <string name="Searched_user">User found by searching for their email address</string>
```
Should start with lowercase

mantasjasikenas commented on app/src/main/res/values/strings.xml:
```diff
@@ -179,4 +179,21 @@
+    <string name="other_users_who_have_sent_you_friend_requests">Other users who have sent you friend requests</string>
+    <string name="search_for_a_user_to_display_them_here">Search for a user to display them here.</string>
+    <string name="Searched_user">User found by searching for their email address</string>
+    <string name="Searched_users">Searched user</string>
```
Should start with lowercase

mantasjasikenas commented on app/src/main/res/values/strings.xml:
```diff
@@ -179,4 +179,21 @@
+    <string name="Searched_user">User found by searching for their email address</string>
+    <string name="Searched_users">Searched user</string>
+    <string name="user_not_found">User not found</string>
+    <string name="Already_friends">You are already friends with this user</string>
```
Should start with lowercase

mantasjasikenas commented on app/src/main/res/values/strings.xml:
```diff
@@ -179,4 +179,21 @@
+    <string name="Searched_users">Searched user</string>
+    <string name="user_not_found">User not found</string>
+    <string name="Already_friends">You are already friends with this user</string>
+    <string name="Self_friend">Can\'t send a friend request to yourself!</string>
```
Should start with lowercase

mantasjasikenas commented on app/src/main/java/com/github/k409/fitflow/ui/screen/friends/FriendRequestScreen.kt:
```diff
@@ -0,0 +1,190 @@
+                        null
+                    }
+                },
+                label = { Text("Send a friend request to") },
```
Move text to string resources

mantasjasikenas commented on app/src/main/java/com/github/k409/fitflow/ui/screen/friends/FriendsScreen.kt:
```diff
@@ -0,0 +1,77 @@
+                .verticalScroll(scrollState)
+                .padding(vertical = 16.dp)
+                .padding(bottom = 6.dp),
+            verticalArrangement = Arrangement.Center,
```
Arrange elements in top of container (verticalArrangement = Arrangement.Top)

mantasjasikenas commented on app/src/main/java/com/github/k409/fitflow/ui/screen/friends/FriendRequestScreen.kt:
```diff
@@ -0,0 +1,190 @@
+}
+
+@Composable
+fun FriendRequestContent(
```
Arrange elements in top of container

GillWell committed:
Update friend invites

Fixed the arrangement to top instead of center

String resource names changed to lowercase

GillWell committed:
Merge branch 'dev' into SGA-128-Friends-invites

GillWell committed:
🤖 Apply Spotless

mantasjasikenas commented:
Please add missing strings in the Lithuanian language

GillWell committed:
Added LT strings

paulius-mongirdas commented on app/src/main/java/com/github/k409/fitflow/ui/screen/friends/FriendCard.kt:
```diff
@@ -0,0 +1,346 @@
+                    color = MaterialTheme.colorScheme.primary,
+                )
+
+                Button(
+                    modifier = Modifier
+                        .padding(start = 6.dp),
+                    colors = ButtonDefaults.buttonColors(MaterialTheme.colorScheme.error),
+                    onClick = {
+                        coroutineScope.launch {
+                            friendsViewModel.removeFriend(user.uid)
+                        }
+                        SnackbarManager.showMessage(context.getString(R.string.friend_removed))
+                    },
```
Maybe add confirmation dialog (like in market purchases) if user really wants to remove friend.

paulius-mongirdas commented on app/src/main/java/com/github/k409/fitflow/ui/screen/friends/FriendsListScreen.kt:
Friends list is not refreshed when friend request is accepted. This seems to be an issue related to device API level, as it works fine on API 34, but does not on 29.

GillWell committed:
Added a dialog to remove friend

GillWell committed:
🤖 Apply Spotless

paulius-mongirdas commented on app/src/main/res/values-lt/strings.xml:
```diff
@@ -289,4 +289,5 @@
     <string name="info_app_usage">Norint užtikrinti, kad jūsų tikslai būtų pasiekti ir jūsų sveikata būtų optimizuota, būtina <b>susieti Health Connect</b> su kitomis programėlėmis, tokiomis kaip Google Fit ar Samsung Health. Be sinchronizacijos, jūsų tikslai gali būti nepasiekti, o bendra jūsų sveikata gali būti pavojinga. \n\nĮsitikinkite, kad <b>įdiegėte Health Connect</b> ir <b>susiejote ją</ ...
     <string name="health_connect">Health Connect</string>
     <string name="info_health_connect"><b>Health Connect</b> yra programėlė, skirta centralizuoti ir racionalizuoti sveikatos duomenis iš įvairių šaltinių, leidžianti vartotojams efektyviau stebėti ir valdyti savo sveikatos rodiklius. Ji veikia kaip mazgas, integruojantis duomenis iš skirtingų sveikatos ir fitneso programėlių, suteikiantis vartotojams išsamų savo sveikatos ir fitneso kelionės vai ...
+    <string name="sure_to_remove_friend">Ar jūs tikrai norite pašilinti draugą </string>
```
Spelling

GillWell committed:
Update strings.xml

GillWell committed:
Merge branch 'SGA-128-Friends-invites' of https://github.com/PVP-K409/fit-flow into SGA-128-Friends-invites
